execution/executor-service/src/thread_executor_service.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{remote_executor_service, remote_executor_service::RemoteExecutorService};
+use aptos_config::utils;
+use std::{
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    thread::{self, JoinHandle},
+};
+
+/// This is a simple implementation of RemoteExecutorService that runs the executor service in a
+/// separate thread. This should be used for testing only.
```

Comment:
> This should be used for testing only.
was wondering why we needed this. Let's put it under the tests? or explicitly mark it testonly